core/src/test/java/org/infinispan/test/fwk/JGroupsConfigBuilder.java

```diff
@@ -146,35 +153,40 @@ private static void removeRelay2(JGroupsProtocolCfg jgroupsCfg) {
       jgroupsCfg.removeProtocol(TEST_RELAY2);
    }
 
-   private static void replaceMCastAddressAndPort(JGroupsProtocolCfg jgroupsCfg, String fullTestName) {
+   private static void replaceMCastAddressAndPort(JGroupsProtocolCfg jgroupsCfg, String fullTestName, int siteIndex) {
       ProtocolConfiguration udp = jgroupsCfg.getProtocol(UDP);
       if (udp == null) return;
 
       Integer udpIndex = testUdpIndex.computeIfAbsent(fullTestName, k -> threadUdpIndex.get());
 
+      if (siteIndex < 0) {
+         siteIndex = 0;
+      }
+      int clusterOffset = udpIndex * MAX_SITES_PER_THREAD + siteIndex;
+
       Map<String, String> props = udp.getProperties();
-      props.put("mcast_addr", "228.10.10." + udpIndex);
-      props.put("mcast_port", String.valueOf(46000 + udpIndex));
+      props.put("mcast_addr", "228.10.10." + clusterOffset);
+      props.put("mcast_port", String.valueOf(46000 + clusterOffset));
       replaceProperties(jgroupsCfg, props, UDP);
    }
 
-   private static void replaceTcpStartPort(JGroupsProtocolCfg jgroupsCfg, TransportFlags transportFlags) {
+   private static void replaceTcpStartPort(JGroupsProtocolCfg jgroupsCfg, int siteIndex) {
       ProtocolType transportProtocol = jgroupsCfg.transportType;
       if (transportProtocol != TCP && transportProtocol != TCP_NIO2)
          return;
 
       Map<String, String> props = jgroupsCfg.getProtocol(transportProtocol).getProperties();
-      Integer startPort = threadTcpStartPort.get();
-      int portRange = TCP_PORT_RANGE_PER_THREAD;
-      if (transportFlags.isPortRangeSpecified()) {
-         portRange = 25;
-         int maxIndex = TCP_PORT_RANGE_PER_THREAD / portRange - 1;
-         if (transportFlags.portRange() > maxIndex) {
-            throw new IllegalStateException("Currently we only support " + (maxIndex + 1) + " ranges/sites!");
-         }
-         startPort += transportFlags.portRange() * portRange;
+      Integer threadIndex = threadTcpIndex.get();
+      int startPort;
+      int portRange;
+      if (siteIndex >= 0) {
```

Comment:
nit
```suggestion
      if (siteIndex > 0) {
```

Reply:
This is intentional, so we use a smaller port range when siteIndex is set to 0 and the full port range when it's not set.